cpp/src/io/device_type.cpp

```diff
@@ -0,0 +1,106 @@
+/*
+ * Copyright (c) 2021, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "cucim/io/device_type.h"
+
+#include <algorithm>
+
+
+namespace cucim::io
+{
+
+// https://en.cppreference.com/w/cpp/algorithm/find
+#if __cplusplus < 202002L
+template <class InputIt, class UnaryPredicate>
+constexpr InputIt myfind_if(InputIt first, InputIt last, UnaryPredicate p)
```

Comment:
Seems like we had this earlier. Should we refactor this code if it is needed? Or if `constexpr` is not needed, we can use from STL. If we do need `constexpr`, maybe Boost also has this and we can use from there?

Reply:
Refactored out as `cucim/cpp/include/cucim/cpp20/find_if.h` file. and reuse the header now.
Thank you for pointing out!